### ISSUE TITLE
url-0.5.9: failed to parse manifest

### ISSUE THREAD

**Issue by krig** [alexcrichton/cargo-vendor#6]
Trying to vendor cargo-vendor itself, I'm getting this error:

```
$ cargo vendor --verbose
 Downloading pnacl-build-helper v1.4.10
 Downloading gdi32-sys v0.2.0
error: unable to get packages from source

Caused by:
  failed to parse manifest at `/home/krig/.cargo/registry/src/github.com-88ac128001ac3a9a/url-0.5.9/Cargo.toml`

Caused by:
  could not parse input as TOML
/home/krig/.cargo/registry/src/github.com-88ac128001ac3a9a/url-0.5.9/Cargo.toml:14:3 expected a newline after table definition

error: third party subcommand `cargo-vendor` exited unsuccessfully

To learn more, run the command again with --verbose.
```

**Comment by krig** [comment]
`cargo build` succeeds without complaint, so there's somehow a difference in how it parses TOML, because it does download and compile url-0.5.9 as well.

**Comment by alexcrichton** [comment]
Ah yeah this is fixed in more recent versions of Cargo (as a library), and this can probably just be fixed by updating dependencies

**Comment by krig** [comment]
The strange thing is that I build cargo version 0.10.0 and it works fine there, and when I build cargo-vendor I also build against cargo 0.10.0. The additional dependencies are exactly the same as for cargo. So it's not quite that simple..

**Comment by krig** [comment]
Something like rust-lang/cargo#2680 is probably the fix that is needed here.

**Comment by krig** [comment]
The odd thing is that I can't find a version of cargo which implements this behaviour. Before using toml 0.1.29, cargo would allow the malformed TOML. After 0.1.29, cargo sets require_newline_after_bool to false and warns about it. I'm clearly missing something here (still a rust newbie) so I think I'm going to have to wait for you to fix it ;)

**Comment by krig** [comment]
Apologies for the spam here, but for the record I figured it out: `Cargo.lock` in cargo-vendor has `toml 0.1.30` whereas `Cargo.lock` in the 0.10.0 branch of cargo has `toml 0.1.28`. Mystery solved, and the fix is simply to roll back to `0.1.28` in `cargo-vendor` until the next version of cargo is released.
